pkg/exthttp/hedging.go

```diff
@@ -0,0 +1,73 @@
+package exthttp
+
+import (
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/caio/go-tdigest"
+	"github.com/cristalhq/hedgedhttp"
+)
+
+type CustomBucketConfig struct {
+	HedgingConfig HedgingConfig `yaml:"hedging_config"`
+}
+
+type HedgingConfig struct {
+	UpTo uint `yaml:"up_to"`
+}
+
+var HedgedOptions CustomBucketConfig
+
+type durationRoundTripper struct {
+	Transport http.RoundTripper
+	TDigest   *tdigest.TDigest
+	mu        sync.Mutex
+}
+
+func (drt *durationRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	start := time.Now()
+	resp, err := drt.Transport.RoundTrip(req)
+	if err != nil {
+		return nil, err
+	}
+	duration := time.Since(start).Seconds()
```

Comment:
Having the delay in milliseconds might be better for overall accuracy?

Reply:
Yes it will be better.